apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ExecuteHelper.cs

```diff
@@ -16,44 +16,42 @@
 using System;
 using System.Linq;
 using System.Threading.Tasks;
+using Google.Api.Gax;
 
 namespace Google.Cloud.Spanner.V1
 {
     internal static class ExecuteHelper
     {
-        public static Task<T> WithSessionChecking<T>(this Task<T> task, Func<Session> sessionFunc)
+        public static async Task<T> WithSessionChecking<T>(this Task<T> task, Func<Session> sessionFunc)
         {
-            return task.ContinueWith(t =>
+            try
             {
-                if (t.IsFaulted)
+                return await task.ConfigureAwait(false);
+            }
+            catch (RpcException ex)
```

Comment:
If you use an exception filter here that always returns false, it preserves stack traces a little better. Can definitely be done later though. (It will also reduce the code duplication between these methods, coincidentally.)